### ISSUE TITLE
Notability iCloud sync unable to turn on

### ISSUE THREAD

**Issue by kenthinson** [openzfsonosx/zfs#309]
Hello. I am unable to turn on Notability iCloud sync when my home folder is on ZFS. When it is on HFS it works fine. I get the error. iCloud isn't Responding. Notability couldn't turn iCloud on because iCloud is taking too long to reply. Please restart your iMac. However restarting does not resolve the issue.

**Comment by brendonhumphrey** [comment]
please provide configuration (hardware), disks layout, o3x version etc etc. Not enough information supplied.

**Comment by ilovezfs** [comment]
Please give this version a try: https://www.dropbox.com/s/n9k9a07yng2x0nh/kenthinson.pkg
It's the current HEAD of spl (54857586823c75d4df389f952b6993738d407a6e) and of zfs (e8725c3149091cf8e50596482d0ad5094135141e).

**Comment by kenthinson** [comment]
Ok sorry here is more information.
I was running the newest master using the compile instructions on the wiki. This allowed iCloud drive to log in and I was able to save with text edit to it. However, I was unable to activate iCloud drive for Notability. 

I uninstalled the build I had made and installed the pkg provided. Now I am unable to access iCloud drive with any app. 

Attached is a screen shot of how it looks now that I am running the pkg. It's turned on but thinks it needs to be upgraded however I have already upgraded my account to iCloud drive. 
![screen shot 2015-05-14 at 1 49 05 pm](https://cloud.githubusercontent.com/assets/2245494/7641727/77bdb5a8-fa40-11e4-8128-086d7ac4a36e.png)

**Comment by kenthinson** [comment]
I have uninstalled the pkg file provided and reinstalled from source using wiki instructions. Now I get this:
![screen shot 2015-05-14 at 1 59 47 pm](https://cloud.githubusercontent.com/assets/2245494/7641897/9fba17f8-fa41-11e4-80b1-7b59c9668d51.png)

As you can see I am able to actually use iCloud drive now. However not with all apps just some.

**Comment by ilovezfs** [comment]
With your own source install re-installed, what is the output of
```
sysctl -a | grep kext_
```

**Comment by kenthinson** [comment]
spl.kext_version: 1.3.1-12_g5485758
zfs.kext_version: 1.3.1-24_ge8725c3

**Comment by brendonhumphrey** [comment]
have you ever managed to have iCloud drive share files into the cloud.

**Comment by kenthinson** [comment]
@brendonhumphrey Great question. In fact I had not tested it on HFS in awhile even though I knew it worked. So i created a new user on HFS too see if it would sync all the same data. I dont usually use the "drive" feature in finder I use other iCloud enabled apps like notability so I had to look in Mobile Documents with terminal to see what was syncing. Long story short the HFS volume showed alot of iCloud data. The ZFS user shows very little data. If i had to guess none of the data on the ZFS user was actually pulled down from iCloud instead it was created by the account.

So long story short no it seems I can place documents in the "iCloud Drive" but nothing is syncing when on ZFS.

**Comment by brendonhumphrey** [comment]
Thats consistent with what I have observed.

I appear to be able to sync photos, notes, maybe some other things, but thats about it.

- Brendon

**Comment by claui** [comment]
Very late to the party; I’d still like to share my experience though as I ran into this issue earlier today on 10.12.6 using ZFS 1.6.1 – right after I had migrated my home directory to a new drive, zpool, and dataset.

Not only did all my iCloud Drive apps fail to sync my documents but those apps also started to freeze erratically; notably, Console.app reported that the `bird` process (which is part of `CloudDocsDaemon.framework`) kept crashing several times per minute with the following message:

```
Assertion failed: (Still unable to open account after attempted reset: <NSError:0x7fe76f5363b0(NSPOSIXErrorDomain:19) - {
    NSDescription = "Operation not supported by device";
}> (underlying: (null), sql: (null))
```

The usual remedies wouldn’t help, such as clearing iCloud’s cache, disabling/re-enabling iCloud Drive, and rebooting.

Turns out that unlike my old dataset, the new dataset didn’t have the `com.apple.mimic_hfs` property set to `on`.
**The moment I ran** `sudo zfs set com.apple.mimic_hfs=on `_`<rootdataset>`,_ **everything worked again.**

I’ve added iCloud Drive to the [list of misbehaving software](https://openzfsonosx.org/w/index.php?title=FAQ&oldid=1232#Q.29_Do_HFS-only_applications_such_as_Photos.2C_iCloud.C2.A0Drive_and_macOS_Server_work_on_ZFS.3F) in the FAQ.
